MSAL/src/native_auth/public/MSALNativeAuthPublicClientApplication.swift

```diff
@@ -49,8 +49,11 @@ public final class MSALNativeAuthPublicClientApplication: MSALPublicClientApplic
 
     let controllerFactory: MSALNativeAuthControllerBuildable
     let inputValidator: MSALNativeAuthInputValidating
-    private let internalChallengeTypes: [MSALNativeAuthInternalChallengeType]
 
+    private(set) static var sharedConfiguration: MSALPublicClientApplicationConfig?
```

Comment:
We can't have static classes like this, as this prevents multiple SDK instances from running in an application in parallel. Since 1 app can only have 1 user flow, and a user flow can only have 1 auth method, we have always said that we support developers from using multiple apps (to support multiple user flows), meaning they need to have multiple SDK instances running in parallel.

cc @nilo-ms, as this an important tenet.

Reply:
Great catch, I believe we all overlooked this. @borgesmb, I think we can use a global variable instead of a `static` one. 

Can you test if this would work? @nilo-ms do you agree with this approach?

Reply:
Just pushed some changes based on an excellent tip from @spetrescu84. We are able to achieve the result without shared properties or spreading references of the client app around.

Reply:
Thanks a lot Sammy for this suggestion! We completely missed it. This static configuration could create weird behaviour even when multiple nativeAuthPublicApplication instances are created (like in our sample app). 
We fixed this issue, I'll let @borgesmb explains the details :)